types/algoliasearch-helper/index.d.ts

```diff
@@ -0,0 +1,891 @@
+// Type definitions for algoliasearch-helper 2.26
+// Project: https://community.algolia.com/algoliasearch-helper-js/
+// Definitions by: Gordon Burgett <https://github.com/gburgett>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+import { Client } from 'algoliasearch';
+import { EventEmitter } from 'events';
+
+/**
+ * The algoliasearchHelper module is the function that will let its
+ * contains everything needed to use the Algoliasearch
+ * Helper. It is a also a function that instanciate the helper.
+ * To use the helper, you also need the Algolia JS client v3.
+ * @param client an AlgoliaSearch client
+ * @param index the name of the index to query
+ * @param opts
+ */
+declare function algoliasearchHelper(client: Client, index: string, opts: Partial<algoliasearchHelper.SearchParameters>): algoliasearchHelper.AlgoliaSearchHelper;
+
+declare namespace algoliasearchHelper {
+  export const version: string;
+
+  export class AlgoliaSearchHelper extends EventEmitter {
+    state: SearchParameters;
+    lastResults: SearchResults;
+    derivedHelpers: AlgoliaSearchHelper[];
+
+    on(event: 'change' | 'search', cb: (state: SearchParameters, lastResults: SearchResults | null) => any): this;
+    on(event: 'searchForFacetValues', cb: (state: SearchParameters, facet?: string, query?: string) => any): this;
```

Comment:
As discussed [here](http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html), it's not recommended to have optional parameters in callbacks.